instance/models/openedx_instance.py

```diff
@@ -117,12 +117,21 @@ def get_load_balancer_configuration(self, triggered_by_instance=False):
         for appserver in active_appservers:
             server_name = "appserver-{}".format(appserver.pk)
             ip_address = appserver.server.public_ip
+            if not ip_address:
+                appserver.logger.error(
+                    "Active appserver does not have a public IP address. This should not happen."
+                    "Calling update_status()"
+                )
+                appserver.server.update_status()
+                ip_address = appserver.server.public_ip
```

Comment:
:+1: Looks good.

If we introduce raising an exception instead of returning `[], []` in the case that the IP is still not available, that exception should probably go right here in this conditional block. Then later below we can erase the extra `if ip_address` check.